### ISSUE TITLE
unable to find the log10() function

### ISSUE THREAD

**Issue by enriqueartal** [gap-packages/anupq#28]
I am trying to compile anupq in Fedora 27, and I got this error. The system is 64 bytes. This is the last part of the output of ./configure

checking for GAP's gmp.h location... not found, GAP was compiled without GMP
checking gmp.h usability... no
checking gmp.h presence... yes
configure: WARNING: gmp.h: present but cannot be compiled
configure: WARNING: gmp.h:     check for missing prerequisite headers?
configure: WARNING: gmp.h: see the Autoconf documentation
configure: WARNING: gmp.h:     section "Present But Cannot Be Compiled"
configure: WARNING: gmp.h: proceeding with the compiler's result
configure: WARNING:     ## ----------------------------------------------------------- ##
configure: WARNING:     ## Report this to https://github.com/gap-packages/anupq/issues ##
configure: WARNING:     ## ----------------------------------------------------------- ##
checking for gmp.h... no
checking for stdlib.h... (cached) yes
checking for GNU libc compatible malloc... no
checking for stdlib.h... (cached) yes
checking for GNU libc compatible realloc... no
checking for pid_t... no
checking vfork.h usability... no
checking vfork.h presence... no
checking for vfork.h... no
checking for fork... no
checking for vfork... no
checking for sys/wait.h that is POSIX.1 compatible... no
checking for gethostname... no
checking for waitpid... no
checking for wait4... no
checking for strftime... no
checking for library containing log10... no
configure: error: unable to find the log10() function

Thanks, Enrique.

**Comment by fingolfin** [comment]
What version of GAP? How did you install it? What version of anupq?

**Comment by enriqueartal** [comment]
The last git version of gap, and for anupq both the git and the version 3.2

**Comment by fingolfin** [comment]
Sorry for neglecting this issue, it got lost in the shuffle.

So, something is completely broken there. The configure log also says that you have no compatible malloc, fork, vfork, etc.

Perhaps you specified some custom CPPFLAGS, CFLAGS, LDFLAGS etc.? That could have messed up compilation. So, please report the exact commands you entered to start `configure`, and perhaps the output of `printenv`. Also, please attach the `config.log` file produced by `configure`.

**Comment by enriqueartal** [comment]
Do not worry! In fact it seems that eventually you neglected the reply but not the issue. After a git pull, it has worked. Thanks, Enrique.

**Comment by wagh** [comment]
I am facing exactly the same problem (error):
I have the following setup:
- Ubuntu - 18.10 (64bit), 
- `GAP` version 4.10.0
- `anupq` version 3.2
- `gcc` - (Ubuntu 8.2.0-7ubuntu1) 8.2.0

Is there any Ubuntu package required? 

-- Vinay

**Comment by fingolfin** [comment]
I'll invest

**Comment by fingolfin** [comment]
I pushed a fix to the master branch. Let me know if this does not work for you.

I'll try to make a new release of this package in time for GAP 4.10.1.

**Comment by fingolfin** [comment]
Odd. Log10 is part of POSIX and of C89. It ought to be there.

**Comment by wagh** [comment]
Some google searches are recommending to `-lm` while compiling. I am not where to use! I can see `-lm` in the Makefile, as a part of `GAP_LIBS`.

My setup is as follows:

- Ubuntu - 19.04 (64bit) (developmental version)
- GAP version 4.10.0
- anupq version 3.2
- gcc - (Ubuntu 8.2.0-12ubuntu2) 8.2.0

 Am I missing any package / library on my system?

**Comment by fingolfin** [comment]
It's no surprise that libm is needed (and hence `-lm`), as that is what the `log10` man page on Linux says. In fact, the test I removed from `configure.ac` last night explicitly tested for whether `libm` is needed, and yet that very tested for some reason failed. I've now added it back, as at least for me, it is needed for make things work on Ubuntu 16.04. It's still mysterious to me why that configure test would fail for some people.

**Comment by fingolfin** [comment]
Since I cannot reproduce it, perhaps those of you who *can* reproduce it could attach their `config.log` file to this issue, those might contain some helpful clues.

**Comment by fingolfin** [comment]
@wagh I am confused now: you write that you are using anupq 3.2 -- but the test I was talking about of course is not in anupq 3.2, it is only in this repository. Could you please clarify which version you tested, exactly?

**Comment by aebr** [comment]
Compilation with -m32 does not work here (x86_64-pc-linux-gnu-default64).
But configure forces -m32. In fact configure.ac contains
```
dnl
dnl anupq require a 32bit compiler, so force that.
dnl
dnl Note that using -m32 like this is of course not universally portable;
dnl but for now, all systems and compilers I know and care about do support it.
dnl If this causes problems for you, please report them via the issue tracker,
dnl including information about your operating system and compiler.
dnl
CFLAGS="$CFLAGS -m32"
LDFLAGS="$LDFLAGS -m32"
```

**Comment by fingolfin** [comment]
@aebr please try the master branch of this repository instead.

**Comment by aebr** [comment]
```./autogen.sh; ./configure; make```
goes fine, problems with gmp.h and log10 disappear now that -m32 is gone (so that conftest works)

**Comment by fingolfin** [comment]
I believe all of this is fixed in anupq 3.2.1 which I just released, and which should be included in the upcoming GAP 4.11.
